src/db/db.erl

```diff
@@ -0,0 +1,19 @@
+-module(db).
+
+-export([query/2]).
+
+query(Q, Params) ->
+    case pgapp:equery(Q, Params) of
+        {ok, _, Columns, Values} -> results_to_map(Columns, Values);
+        {ok, Columns, Values} -> results_to_map(Columns, Values);
+        E -> {error, E}
+    end.
+
+results_to_map(ColumnNames, RowList) ->
+    Keys = [Name || {column, Name, _, _, _, _} <- ColumnNames],
+    ToMap = fun (Row) ->
+                    RowAsList = tuple_to_list(Row),
+                    maps:from_list(lists:zip(Keys, RowAsList))
```

Comment:
But those are two different lists, the comprehenssion traverses the column names list (to extract the names), the fun is mapping over the list of rows